File: webserver/app/utils/general/sanitize.py
import re


def sanitize_email(unsafe_email: str) -> str:
    """
    Helper function for sanitizing an email
    Args:
        unsafe_email (str): the unsanitized email
    Returns:
        str: the received email if it's valid
        an empty string if it's invalid
    """

    email_regex = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
    if re.fullmatch(email_regex, unsafe_email):
        return unsafe_email
    else:
        return ""

File: backend/services/scaling-service/algorithms/helpers/sanitize.go
package helpers

import (
	"regexp"

	"github.com/whisthq/whist/backend/services/utils"
)

// SanitizeEmail tries to match an email to a general email regex
// and if it fails, returns an empty string. This is done because
// the email can be spoofed from the frontend.
func SanitizeEmail(email string) (string, error) {
	emailRegex := `(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)`

	match, err := regexp.Match(emailRegex, []byte(email))
	if err != nil {
		return "", utils.MakeError("error sanitizing user email. Err: %v", err)
	}

	if match {
		return email, nil
	} else {
		return "", nil
	}
}
